File: Problems 2000-2099/2099. Find Subsequence of Length K With the Largest Sum/solution.ts
function maxSubsequence(nums: number[], k: number): number[] {
    const res: number[] = nums.slice(0, k);

    for (let i = k; i < nums.length; i++) {
        const currMin = Math.min(...res);
        if (nums[i] > currMin) {
            res.splice(res.indexOf(currMin), 1);
            res.push(nums[i]);
        }
    }

    return res;

    // const res = nums.map((num, i) => [num, i])
    //             .sort((a, b) => b[0] - a[0])
    //             .slice(0, k)
    //             .sort((a, b) => a[1] - b[1])
    //             .map(el => el[0]);
    // return res;
};

File: Problems 2000-2099/2099. Find Subsequence of Length K With the Largest Sum/solution.py
class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        indexed = [(num, i) for i, num in enumerate(nums)]

        top_k = sorted(indexed, key=lambda x: x[0], reverse=True)[:k]

        top_k.sort(key=lambda x: x[1])

        return [num for num, _ in top_k]

        # res = nums[:k]

        # for i in range(k, len(nums)):
        #     currMin = min(res)
        #     if nums[i] > currMin:
        #         res.remove(currMin)
        #         res.append(nums[i])
        
        # return res